hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -374,15 +375,18 @@ def fill_for(elements, unique, fill, name=""):
     return fill
 
 
+D = TypeVar("D", bound=DTypeLike)
+
+
 @defines_strategy(force_reusable_values=True)
 def arrays(
-    dtype: Any,
+    dtype: D,
```

Comment:
This can actually be a `DTypeLike` *or* a `SearchStrategy[DTypeLike]`, so we might need an overload to express it properly.

Reply:
Ok, I guess writing this as `Union[dtype: D, SearchStrategy[DTypeLike]]` should do then, I'll try.